src/item_contents.cpp

```diff
@@ -347,21 +348,38 @@ void item_contents::heat_up()
     }
 }
 
-int item_contents::ammo_consume( int qty )
+int item_contents::ammo_consume( int qty, const tripoint &pos )
 {
     for( item_pocket &pocket : contents ) {
-        if( !pocket.is_type( item_pocket::pocket_type::MAGAZINE ) ) {
-            continue;
+        if( pocket.is_type( item_pocket::pocket_type::MAGAZINE_WELL ) ) {
+            // we are assuming only one magazine per well
```

Comment:
This assumption is fine to bake into the code. There should only ever be one magazine per well. Going forward we should handle multiple magazines with multiple `MAGAZINE_WELL` type pockets.